scripts/db_migrator.py

```diff
@@ -1277,7 +1276,14 @@ def main():
         socket_path = args.socket
         namespace = args.namespace
 
-        load_db_config()
+        # Can't load global config base on the resoult of is_multi_asic(), because on multi-asic device, when db_migrate.py
+        # run on the local database, ASIC instance will have not created the /var/run/redis0/sonic-db/database-config.json
+        if args.namespace is not None:
+            if not SonicDBConfig.isGlobalInit():
+                SonicDBConfig.load_sonic_global_db_config(namespace=args.namespace)
+        else:
+            if not SonicDBConfig.isInit():
+                SonicDBConfig.load_sonic_db_config()
 
```

Comment:
I feel it is more like a timing issue as you pointed out, Can we call db_mibrator, after /var/run/redis0/sonic-db/database-config.json is created ? 

Here, https://github.com/sonic-net/sonic-swss-common/blob/7db6ccf5fb427d4832d6c54ab1f485ddd87b6697/common/dbconnector.h#L107 -- it doesn't make a difference if we pass/don't pass the namespace parameter to load_sonic_global_db_config()

Reply:
I search sonic code but not found any code use the '-n' parameter of db_migrator.py to pass the namespace parameter.
I will add comments in the https://github.com/sonic-net/sonic-buildimage/issues/18389 to ask author check if this can fix by your suggestion.

For load_sonic_global_db_config(namespace=args.namespace), I think it's better keep the parameter to keep the compatibility, incase we change the code of load_sonic_global_db_config in future.

Reply:
@liuh-80 @judyjoseph  all instance database@.service have dependency "Requires=database.service
After=database.service".  db_mihgrator script in local host database will not be able to wait for all other instance ready  unless we change the dependency.